### PR TITLE
Enforce autentication for swagger endpoints

### PR DESCRIPTION
**Pull request by OferLevi85**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSBinder.java [hunk 1]
```diff
@@ -21,6 +21,6 @@
 
     @Override
     protected void configure() {
-        bind(new ZMSImpl()).to(ZMSHandler.class);
+        bind(ZMSImplFactory.getZmsInstance()).to(ZMSHandler.class);
```
Comment (OferLevi85) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSBinder.java:24]: I use the same zms instance to check authentication.

File: servers/zms/src/main/java/com/yahoo/athenz/zms/servlet/SwaggerFilter.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    private final ZMSHandler zmsHandler = ZMSImplFactory.getZmsInstance();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        ResourceContext context = this.zmsHandler.newResourceContext((HttpServletRequest) request, (HttpServletResponse) response, "swagger");
```
Comment (OferLevi85) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/servlet/SwaggerFilter.java:27]: Basically the same check for authentication as in ZMSResources / ZTSResources

File: servers/zms/src/main/webapp/WEB-INF/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+prettyPrint: true
```
Comment (OferLevi85) [inline servers/zms/src/main/webapp/WEB-INF/openapi.yaml:1]: A configuration file for swagger. Determines how the yaml will be displayed.

File: servers/zms/src/main/webapp/WEB-INF/web.xml [hunk 4]
```diff
@@ -31,9 +32,20 @@
         <load-on-startup>2</load-on-startup>
     </servlet>
     <servlet-mapping>
-        <servlet-name>jersey</servlet-name>
+        <servlet-name>swagger-servlet</servlet-name>
         <url-pattern>/v1/api/*</url-pattern>
     </servlet-mapping>
+
+    <!-- Authentication filter for swagger -->
```
Comment (OferLevi85) [inline servers/zms/src/main/webapp/WEB-INF/web.xml:39]: Register the authentication filter for swagger servlet... same for ZTS

**Comment by abvaidya** [comment]
Please add copyright section in new files.

**Comment by OferLevi85** [comment]
> Please add copyright section in new files.
Fixed

**Review by havetisyan — APPROVED** [review]
👍

**Pull request merged**